onnxruntime/core/framework/session_options.cc

```diff
@@ -96,4 +97,22 @@ void SessionOptions::AddCustomOpLibraryHandle(PathString library_name, void* lib
 }
 #endif  // !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)
 
+EpContextModelGenerationOptions SessionOptions::GetEpContextGenerationOptions() const {
+  if (this->has_explicit_ep_context_gen_options) {
+    return this->ep_context_gen_options;
+  }
+
+  // Have to generate a struct from session config options.
+  EpContextModelGenerationOptions options{};
+  options.enable = this->config_options.GetConfigOrDefault(kOrtSessionOptionEpContextEnable, "0") == "1";
+  options.output_model_file_path = this->config_options.GetConfigOrDefault(kOrtSessionOptionEpContextFilePath, "");
+  options.output_external_initializers_file_path = this->config_options.GetConfigOrDefault(
+      kOrtSessionOptionsEpContextModelExternalInitializersFileName, "");
+  options.output_external_initializer_size_threshold = 0;
+  options.embed_ep_context_in_model = this->config_options.GetConfigOrDefault(
+                                          kOrtSessionOptionEpContextEmbedMode, "0") == "1";
```

Comment:
could this be in a ctor for EpContextModelGenerationOptions?

Reply:
Yep, moved into a constructor.